datasource/ethereum/src/ethereum_adapter.rs

```diff
@@ -255,39 +256,61 @@ where
             if start > to {
                 return None;
             }
-            let end = start + step;
-            debug!(logger, "Requesting logs for blocks [{}, {}]", start, end);
+
+            // This will iterate more than once iff `step` has been reduced,
+            // making as many parallel requests of size `step` as necessary,
+            // respecting `LOG_STREAM_PARALLEL_CHUNKS`.
+            let mut chunk_futures = vec![];
+            let mut offset = start;
+            for _ in 0..*LOG_STREAM_PARALLEL_CHUNKS {
```

Comment:
Curious, is there a good way to accumulate the `chunk_futures` with an iterator over the `0..*LOG_STREAM_PARALLEL_CHUNKS` range?

Reply:
I can't think of a very good way of turning this loop into combinators, but maybe there is one.

Reply:
https://docs.rs/itertools/0.3.0/itertools/fn.times.html

Reply:
Basically,
```rust
let chunk_futures = itertools::times(*LOG_STREAM_PARALLEL_CHUNKS)
  .map(|_| { ... })
  .collect();

stream::futures_ordered(chunk_futures)
  ...
```